### ISSUE TITLE
Typelevel compiler support

### ISSUE THREAD

**Issue by cquiroz** [scalacenter/scalafix#397]
I'm testing scalafix in a project that uses the typelevel compiler and I get this warning:

```
[warn] scalafix - Unsupported edu_gemini_seqexec_engine/scalaVersion 2.12.3-bin-typelevel-4. Please upgrade to one of: 2.11.11, 2.12.3
[warn] scalafix - Unsupported seqexec_server_gn_test/scalaVersion 2.12.3-bin-typelevel-4. Please upgrade to one of: 2.11.11, 2.12.3
[warn] scalafix - Unsupported edu_gemini_seqexec_modelJVM/scalaVersion 2.12.3-bin-typelevel-4. Please upgrade to one of: 2.11.11, 2.12.3
...
```

Should this configuration be supported?

**Comment by olafurpg** [comment]
I'm not sure what binary compatibility typelevel promises for compiler plugins. Is the warning spurious or do you get any errors during compilation? The semanticdb-scalac compiler plugin, which scalafix depends on, touches a decent surface of the compiler APIs. 

We could change sbt-scalafix to create a `scalafixSupportedScalaVersions` settingKey and make it configurable. Here is where the warning is reported
https://github.com/scalacenter/scalafix/blob/25503f647b4b66f234e0b7f37bfa78279e83d3a7/scalafix-sbt/src/main/scala/scalafix/sbt/ScalafixPlugin.scala#L159
and here is where the default value is generated  with BuildInfo https://github.com/scalacenter/scalafix/blob/25503f647b4b66f234e0b7f37bfa78279e83d3a7/build.sbt#L129

**Comment by cquiroz** [comment]
It is displayed on startup and everytime I call `scalafix` on the shell. i don't see compilation errors but I've done little testing

**Comment by gabro** [comment]
If it's not blocking (i.e. it's just a warning), I'm in favor of keeping the warning with the Typelevel compiler (which is in fact not officially supported) and maybe rephrase it to:

> Unsupported Scala version (edu_gemini_seqexec_engine/scalaVersion 2.12.3-bin-typelevel-4). If you have issues, consider using one of the supported versions: 2.11.11, 2.12.3

**Comment by cquiroz** [comment]
i got it to work by setting the plugin compiler's version explicitly via
```
addCompilerPlugin("org.scalameta" % "semanticdb-scalac_2.12.4" % "2.1.2")
```

**Comment by olafurpg** [comment]
Thank you for the workaround @cquiroz . Official Typelevel scala support is not on the roadmap but I suspect your workaround will work sufficiently well for people using Typelevel Scala.
